common/mariadb/values.yaml

```diff
@@ -18,10 +18,29 @@ join_buffer_size: "4M"
 binlog_format: "MIXED"
 expire_logs_days: 10
 
+
+pre_change_hook:
+  image: "sapcc/unified-kubernetes-toolbox"
+  image_version: "20220525082919"
```

Comment:
an extra image dependency, I assume the job will be failing on k8s upgrade and we would need extra coordination for deployment of all openstack services...

Reply:
I agree some extra work is needed on adapting the pipeline to make it work more seamlessly.
But I wouldn't say it requires extra coordination.  It uses the same image repository as in the ci-pipeline itself.

We need to update the pipelines (manually) when the image changes. That is exactly also the time we need to change this image version. So no extra coordination, but yes, extra manual effort.

We can change the pipeline-definition(s) to use exactly the same version, when the pipeline is applied.
Or we can change the pipeline to transport the latest version through the pipeline the same way all other images are transported.
Personally, I have a slight preference to the latter, but how it is done is a different discussion, I would say.

Reply:
Just a remark: Not all users of the shared pipeline use image version transporting and not all users of this chart (I would assume) use the shared pipeline. Additionally, people tend to only update the shared tasks for a new image-version and not every reference, so Dmitry has a good point here.

Reply:
> Just a remark: Not all users of the shared pipeline use image version transporting and not all users of this chart (I would assume) use the shared pipeline

Well, but also the hook will only trigger if the "managed" database / user fields are used, so I don't think we have to cover all use-cases. Personally, I am fine with just the "openstack" services, or even nova and ironic, as they have more than one database & schema.

> Additionally, people tend to only update the shared tasks for a new image-version and not every reference, so Dmitry has a good point here.

I do not follow... 

I understand Dmitry the following way:
We upgrade k8s, so we need to ensure that the toolbox used everywhere is current enough.

There are 31 references to the `sapcc/unified-kubernetes-toolbox` in our pipeline definitions, 10 of those are under "shared", except in one location for ironic, all openstack services use the shared ones.

If we *need* to update the references in a pipeline then someone has to manually change those references,
and manually apply the pipeline. 

I am saying, the very procedure could result in the very same image being used the next time the pipeline is triggered inside this hooks as well in the pipeline job. 
Without effort of improving the situation, it just means that one has to change 32 places instead of 31.

Personally, I would opt for transporting the image version through the pipeline, and leave it to the teams which opt-out of image transport, but want to opt-in into the new functionality how they want to do it.